### PR TITLE
Cutout

### PR DESCRIPTION
**Pull request by tbennun**

### CODE REVIEW

File: src/utils/sdfg/json_serializer.ts [hunk 1]
```diff
@@ -23,10 +23,9 @@ function isDict(v: unknown): v is Record<string, unknown> {
         !(v instanceof Date);
 }
 
-function replacer(_name: string, val: unknown, orig_sdfg: unknown): unknown {
-    if (val && isDict(val) && val !== orig_sdfg && 'type' in val &&
-        val.type === 'SDFG') {
-        return JSON.stringify(val, (n, v) => replacer(n, v, val));
```
Comment (phschaad) [inline src/utils/sdfg/json_serializer.ts:29]: What's the rationale with which we can drop this? What was it good for?

File: src/renderer/renderer.ts [hunk 2]
```diff
@@ -57,6 +58,10 @@ declare const canvas2pdf: any;
 // Some global functions and variables which are only accessible within VSCode:
 declare const vscode: any | null;
 
+type SDFGElementType = 'states' | 'nodes' | 'edges' | 'isedges';
+// If type is explicitly set, dagre typecheck fails with integer node ids
+export type SDFGListType = any[];//{ [key: number]: DagreSDFG };
```
Comment (phschaad) [inline src/renderer/renderer.ts:63]: Hm, sad that this can't be further typed. But something to deal with this in the overhault.

**Review by phschaad — COMMENTED** [review]
LGTM, testing with VSCode before I accept

**Review by phschaad — APPROVED** [review]

File: src/utils/sdfg/json_serializer.ts (hunk 1 above)
Comment (tbennun) [reply src/utils/sdfg/json_serializer.ts:29]: It used to be a string, but now we keep it as a sub dictionary. So now we keep a backwards compatibility on reading but not writing.

File: src/renderer/renderer.ts (hunk 2 above)
Comment (tbennun) [reply src/renderer/renderer.ts:63]: Yes

**Pull request merged**